server/src/session.rs

```diff
@@ -1,15 +1,44 @@
 // org.freedesktop.Secret.Session
 
-use zbus::interface;
+use std::sync::{Arc, Mutex};
+
+use oo7::Key;
+use zbus::{interface, zvariant::OwnedObjectPath};
 
 use super::Result;
+use crate::service_manager::ServiceManager;
 
-#[derive(Debug)]
-pub struct Session {}
+#[derive(Debug, Clone)]
+pub struct Session {
+    _aes_key: Option<Arc<Key>>,
+    manager: Arc<Mutex<ServiceManager>>,
+    path: OwnedObjectPath,
+}
 
 #[interface(name = "org.freedesktop.Secret.Session")]
 impl Session {
-    pub async fn close(&self) -> Result<()> {
-        todo!()
+    pub async fn close(
+        &self,
+        #[zbus(object_server)] object_server: &zbus::ObjectServer,
+    ) -> Result<()> {
+        self.manager.lock().unwrap().remove_session(&self.path);
+        object_server.remove::<Self, _>(&self.path).await?;
+
+        Ok(())
+    }
+}
+
+impl Session {
+    pub fn new(aes_key: Option<Arc<Key>>, manager: Arc<Mutex<ServiceManager>>, index: i32) -> Self {
+        Self {
+            path: OwnedObjectPath::try_from(format!("/org/freedesktop/secrets/session/s{index}"))
+                .unwrap(),
+            _aes_key: aes_key,
+            manager,
+        }
+    }
+
+    pub fn path(&self) -> &OwnedObjectPath {
```

Comment:
Does this need to be public?

Reply:
Can it be a zvariant::ObjectPath instead?

Reply:
it is a binary, so it doesn't matter for the public part. And it can actually return an ObjectPath i think